Add permissions cli

tmihalac opened the pull request:
- Added cli command permissions
- Added tags parameter to list_validation_references and list_saved_datasets in registry
- Added list_validation_references and list_saved_datasets apis to feature_store
- Added missing tags parameters to registry_server methods

lokeshrangineni commented on sdk/python/feast/cli.py:
```diff
@@ -870,5 +873,273 @@ def validate(
+                    BatchFeatureView,
+                    StreamFeatureView,
+                ]:
+                    _handle_fv_verbose_permissions_command(
```
can we move these methods separate file so that cli.py will act as a simple controller?

lokeshrangineni commented on sdk/python/feast/cli.py:
```diff
@@ -870,5 +873,273 @@ def validate(
+
+
+def _handle_sd_verbose_permissions_command(
+    feast_type, p, policy_node, store, tags_filter
```
could we mention the types for parameters?

dmartinol commented:
Pls add some sample outputs for verbose and standard mode.
LGTM

redhatHameed commented:
The PR need rebase before merge

tmihalac committed:
CLI command "feast permissions list"
Added cli command permissions
Added tags parameter to list_validation_references and list_saved_datasets in registry
Added list_validation_references and list_saved_datasets apis to feature_store
Added missing tags parameters to registry_server methods

Signed-off-by: Theodor Mihalache <tmihalac@redhat.com>

tmihalac commented:
> Pls add some sample outputs for verbose and standard mode. LGTM

Where do you want me to add it? currently I've added and example in addition actual data when you run `feature_repo % feast feast-permissions list --verbose`, is this good enough ?
<img width="1695" alt="image" src="https://github.com/RHEcosystemAppEng/feast/assets/84387487/891d515c-5861-4aee-98b1-5d2360a6ee65">

tmihalac committed:
CLI command "feast permissions list"
- Changes following review

Signed-off-by: Theodor Mihalache <tmihalac@redhat.com>

dmartinol reviewed (commented):
can you update the user guide in `docs/reference/feast-cli-commands.md`?

tmihalac committed:
CLI command "feast permissions list"
- Changes following review

Signed-off-by: Theodor Mihalache <tmihalac@redhat.com>
